wandb/testing/relay.py

```diff
@@ -309,21 +309,16 @@ def resolve_uploaded_files(
     ) -> Optional[Dict[str, Any]]:
         if not isinstance(request_data, dict) or not isinstance(response_data, dict):
             return None
-        query = "RunUploadUrls" in request_data.get("query", "")
+
+        query = "CreateRunFiles" in request_data.get("query", "")
```

Comment:
maybe it would be better to add another resolver? 
not strictly required, just will allow to test agains older local servers.

Reply:
when you say "add another resolver", is it something like this? if we add these, do we also want to add version checking and condition using one resolver method over the other?

<img width="1577" alt="Screenshot 2023-06-29 at 4 15 57 PM" src="https://github.com/wandb/wandb/assets/69057500/c2757c23-a21c-4e20-912d-eb9df3bad8fc">

Reply:
Katia and Dima explained to me how these tests work — thank you! Leaving some note for my own sake:

- relay.py — a middle person who snoop (with `snoop_context()`) into the request (e.g. graphql, storage, etc) and can modify the results and store the data to context
- Example use case of relay — changing status code from 200 to 400 to test error handling, modifying the results and storing it in context so downstream tests can reuse it from the context easily (this is the use case for upload URL)
- if QueryResolver doesn't have the resolver / request you are working on, that means no snooping happening with that requst, just making and handling the pure request
- `QueryResolver.resolve()` iterates through resolvers registered in the QueryResolver list to see if there is matching request that resolver should handle
- two types of high level test implementations — mock server vs. using real backend from local-testcontainer (more details explained really well with diagram [here](https://www.notion.so/wandbai/Un-mock-my-mock-server-0202bd99f1844ba086ab615bec9f4e57?pvs=4))
- technically, we can handle the raw results from graphql in each test without having `resolve_upload_urls` in `QueryResolver` but there are many tests that use upload URL's so query resolver helps simplify this flow at relay level